shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.toErrorMessage
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import co.nimblehq.blisskmmic.presentation.modules.ViewModel
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
+    val isSuccess: Boolean = false,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    constructor() : this(false, false, null)
+
+    constructor(error: String?) : this(false, false, error)
+}
+
+class LoginViewModel(private val logInUseCase: LogInUseCase): ViewModel() {
+
+    private val mutableViewState: MutableStateFlow<LoginViewState> =
```

Comment:
@blyscuit I think using `SharedFlow` is better, as we do not need to get the default of state 🤔

Reply:
Could you explain more on `SharedFlow`. I thought the main difference between `StateFlow` and `SharedFlow` is cold and hot (respectively). Wouldn't cold state and no relay be better for the ViewModel ViewState?
I don't quite get the picture how can we use `SharedFlow` to `not need to get the default of state`?

Reply:
Stateflow takes an initial value through constructor and emits it immediately when someone starts collecting. So you will receive the default `LoginViewState` at the beginning. I guess we don't need this. It is an optional 🙏

Reply:
From my testing with the iOS app, the original State is correctly receive when subscribe, no need for relay. Let's keep it this way for now, if it doesn't work with Android we can move to SharedFlow. 👍

Reply:
Agree @blyscuit